app/Jobs/ApplicationDeploymentJob.php

```
@@ -229,7 +229,19 @@ public function __construct(public int $application_deployment_queue_id)
 
         // Set preview fqdn
         if ($this->pull_request_id !== 0) {
-            $this->preview = $this->application->generate_preview_fqdn($this->pull_request_id);
+            if ($this->application->build_pack === 'dockercompose') {
+                // For Docker Compose apps, use the preview model's compose-specific method
+                $this->preview = ApplicationPreview::findPreviewByApplicationAndPullId($this->application->id, $this->pull_request_id);
```

PHPMD (2.15.0)

Avoid using static access to class '\App\Models\ApplicationPreview' in method '__construct'. (Clean Code Rules)

(StaticAccess)